scaleway/resource_vpc_gateway_network.go

```diff
@@ -155,15 +169,39 @@ func resourceScalewayVPCGatewayNetworkRead(ctx context.Context, d *schema.Resour
 		return diag.FromErr(err)
 	}
 
-	gatewayNetwork, err := vpcgwNetworkAPI.GetGatewayNetwork(&vpcgw.GetGatewayNetworkRequest{
+	readGWNetwork := &vpcgw.GetGatewayNetworkRequest{
+		GatewayNetworkID: ID,
+		Zone:             zone,
+	}
+
+	retryInterval := retryIntervalVPCGatewayNetwork
+	_, err = vpcgwNetworkAPI.WaitForGatewayNetwork(&vpcgw.WaitForGatewayNetworkRequest{
 		GatewayNetworkID: ID,
+		Timeout:          scw.TimeDurationPtr(defaultVPCGatewayTimeout),
+		RetryInterval:    &retryInterval,
 		Zone:             zone,
 	}, scw.WithContext(ctx))
+	// check err waiting process
 	if err != nil {
-		if is404Error(err) {
-			d.SetId("")
-			return nil
+		return diag.FromErr(err)
+	}
+
+	var gatewayNetwork *vpcgw.GatewayNetwork
+	err = retryVPCContext(ctx, retryGWTimeout, func() *resource.RetryError {
+		currentGW, errReadVPC := vpcgwNetworkAPI.GetGatewayNetwork(readGWNetwork, scw.WithContext(ctx))
+		if isGatewayNetworkMacAddressEmpty(currentGW) {
+			return resource.RetryableError(fmt.Errorf("missing attribute"))
```

Comment:
This message is not clear. Could you put something more explicit about what is going on? For instance: Gateway Network got no mac addresses.